routes/logger.go

```diff
@@ -0,0 +1,69 @@
+package routes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/gocardless/draupnir/routes/chain"
+	"github.com/prometheus/common/log"
+)
+
+type key int
+
+const loggerKey key = 1
+
+func NewRequestLogger(logger log.Logger) chain.Middleware {
```

Comment:
As a side note, I prefer the idea of creating a logger over the previous `LogRequest`